render/gles2: use glGetAttribLocation instead of hardcoded indices

Bl4ckb0ne opened the pull request:
Fixes #2265

ascent12 commented on render/gles2/renderer.c:
```diff
@@ -205,7 +210,20 @@ static void gles2_render_ellipse_with_matrix(struct wlr_renderer *wlr_renderer,
+	glVertexAttribPointer(renderer->shaders.ellipse.tex_loc, 2, GL_FLOAT,
+			GL_FALSE, 0, texcoord);
+
+	glEnableVertexAttribArray(0);
+	glEnableVertexAttribArray(1);
```
Shouldn't these use the *_loc members too?

Bl4ckb0ne replied:
It could yes, do you want me to change it?

ascent12 replied:
Yes.

Bl4ckb0ne replied:
Done for all calls to `glEnableVertexAttribArray`

emersion commented on include/render/gles2.h:
```diff
@@ -56,11 +58,15 @@ struct wlr_gles2_renderer {
 			GLint proj;
 			GLint color;
+			GLint pos_loc;
+			GLint tex_loc;
```
I don't think the ellipse shader uses texcoords

Bl4ckb0ne replied:
The ellipse shader uses [`ellipse_fragment_src`](https://github.com/swaywm/wlroots/blob/master/render/gles2/renderer.c#L720), which uses the [`texcoord`](https://github.com/swaywm/wlroots/blob/master/render/gles2/shaders.c#L35) attribute

emersion replied:
Ah, indeed.

emersion commented on render/gles2/shaders.c:
```diff
@@ -6,14 +6,14 @@ const GLchar quad_vertex_src[] =
 "uniform vec4 color;\n"
 "attribute vec2 pos;\n"
-"attribute vec2 texcoord;\n"
+//"attribute vec2 texcoord;\n"
```
We can probably get rid of this (and the rest of the texcoord stuff) for colored quads

Bl4ckb0ne replied:
Oops, this was from a debug session. But yes, the `pos` vertices and the `texcoord` vertices are the same, I think we could reuse the same array of vertices.

Bl4ckb0ne replied:
This has been put back.

emersion commented on include/render/gles2.h:
```diff
@@ -56,11 +58,15 @@ struct wlr_gles2_renderer {
 			GLint proj;
 			GLint color;
+			GLint pos_loc;
+			GLint tex_loc;
```
This isn't used

Bl4ckb0ne replied:
Done

emersion commented on render/gles2/renderer.c:
```diff
@@ -205,7 +210,20 @@ static void gles2_render_ellipse_with_matrix(struct wlr_renderer *wlr_renderer,
 	glUniform4f(renderer->shaders.ellipse.color, color[0], color[1], color[2], color[3]);
-	draw_quad();
+
+
```
Style: stray newline here

Bl4ckb0ne replied:
Done

emersion commented on render/gles2/renderer.c:
```diff
@@ -86,32 +99,6 @@ static void gles2_scissor(struct wlr_renderer *wlr_renderer,
 	POP_GLES2_DEBUG;
 }
 
-static void draw_quad(void) {
```
I wonder if it's worth it to keep this function and make it take `pos_loc` and `tex_loc`.

emersion replied:
Eh, probably not

emersion commented on render/gles2/renderer.c:
```diff
@@ -656,6 +678,8 @@ struct wlr_renderer *wlr_gles2_renderer_create(struct wlr_egl *egl) {
 	renderer->shaders.tex_rgbx.invert_y = glGetUniformLocation(prog, "invert_y");
 	renderer->shaders.tex_rgbx.tex = glGetUniformLocation(prog, "tex");
 	renderer->shaders.tex_rgbx.alpha = glGetUniformLocation(prog, "alpha");
+	renderer->shaders.quad.pos_loc = glGetAttribLocation(prog, "pos");
```
s/quad/tex_rgbx/

Bl4ckb0ne replied:
Done

emersion commented on include/render/gles2.h:
```diff
@@ -37,6 +37,8 @@ struct wlr_gles2_tex_shader {
 	GLint invert_y;
 	GLint tex;
 	GLint alpha;
+	GLint pos_loc;
```
Nit: the existing members are locations too, so it's inconsistent to add a `_loc` suffix to the new members, we should drop it

Bl4ckb0ne replied:
It's an attribute location compared to other that are uniform locations. `pos_attrib` maybe?

emersion replied:
Works for me

Bl4ckb0ne replied:
Done

emersion commented:
Needs rebase.

Bl4ckb0ne commented:
I rebased and squashed, but I'd like to have another review please, I had to make a few changes caused by the merge of https://github.com/swaywm/wlroots/pull/2092
